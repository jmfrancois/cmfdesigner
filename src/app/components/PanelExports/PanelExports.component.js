import React from 'react';
import PropTypes from 'prop-types';
import { Inject } from '@talend/react-cmf';
import ViewFunctionParam from '../ViewFunctionParam';

class PanelExports extends React.Component {

	constructor(props) {
		super(props);
		this.state = { opened: true };
		this.onClick = this.onClick.bind(this);
	}

	onClick(event) {
		event.preventDefault();
		this.setState({ opened: !this.state.opened });
	}

	render() {
		if (this.props.dependencies.length === 0) {
			return null;
		}
		return (
			<div>
				<h3>
					<a href="#open-export" onClick={this.onClick}>
						<Inject component="Icon" componentId={this.state.opened ? 'selection-list-opened' : 'selection-list-closed'} />
						Exports
					</a>
				</h3>
				{this.state.opened && (
					<ul>
						{this.props.dependencies.map((dep, index) => (
							<li key={index}>
								export [{dep.type}{dep.generator && '*'}] {dep.name} params: <ViewFunctionParam params={dep.params} />
							</li>
						))}
					</ul>
				)}
			</div>
		);
	}
}

PanelExports.propTypes = {
	dependencies: PropTypes.arrayOf(PropTypes.shape({
		default: PropTypes.bool,
		name: PropTypes.string,
	})),
};
PanelExports.defaultProps = {
	dependencies: [],
};
PanelExports.displayName = 'PanelExports';

export default PanelExports;
