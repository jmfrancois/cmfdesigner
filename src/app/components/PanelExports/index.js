import PanelExports from './PanelExports.component';

export default PanelExports;
