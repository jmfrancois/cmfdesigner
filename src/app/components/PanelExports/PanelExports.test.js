import React from 'react';
import { shallow } from 'enzyme';

import PanelExports from './PanelExports.component';

describe('PanelExports', () => {
	it('should render', () => {
		const wrapper = shallow(
			<PanelExports />
		);
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
