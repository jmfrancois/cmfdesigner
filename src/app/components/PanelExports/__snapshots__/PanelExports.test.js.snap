// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelExports should render 1`] = `null`;
